src/main/java/codeu/model/data/User.java

```diff
@@ -23,36 +24,48 @@
   private final String name;
   private final String passwordHash;
   private final Instant creation;
+  private long messageCount;
+  private boolean isAdmin;
   private  String description;
 
-
   /**
    * Constructs a new User.
    *
-   * @param id the ID of this User
-   * @param name the username of this User
+   * @param id           the ID of this User
+   * @param name         the username of this User
    * @param passwordHash the password hash of this User
-   * @param creation the creation time of this User
-   * @param description of the username from profile section
+   * @param creation     the creation time of this User
+   * @param mesageCount  the message count of this User
+   * @param isAdmin      if the user is an admin or not
+   * @param description  of the username from profile section
    */
-  public User(UUID id, String name, String passwordHash, Instant creation, String description) {
+  public User(UUID id, String name, String passwordHash, Instant creation, long messageCount, 
+              boolean isAdmin, String description) {
     this.id = id;
     this.name = name;
     this.passwordHash = passwordHash;
     this.creation = creation;
+    this.messageCount = messageCount;
+    this.isAdmin = isAdmin;
     this.description = description;
   }
 
   /**
+   * Constructs a new User.
    *
-   * @param id the ID of this User
-   * @param name the username of this User
+   * @param id           the ID of this User
+   * @param name         the username of this User
    * @param passwordHash the password hash of this User
-   * @param creation the creation time of this User
-   * @param "" for the user who is writing their description for first time
+   * @param creation     the creation time of this User
+   * @param isAdmin      if the user is an admin or not
    */
-  public User(UUID id, String name, String passwordHash, Instant creation) {
-    this(id, name, passwordHash, creation, "");
+  public User(UUID id, String name, String passwordHash, Instant creation, boolean isAdmin) {
```

Comment:
I still dont think we need the isAdmin parameter. We can computer isAdmin in this User class. So instead of having the RegisterServlet handle it.

Reply:
I tried, but Java didn't let me, it gave me a build failure

Reply:
But I will try another thing

Reply:
This other thing that I believed that could work it didn't. First I wanted to write code before calling the other constructor, but java didn't let me. And the other solution that I thought was using another method to give the value, but also Java doesn't like it

Reply:
I have another idea, more stupid, but I'm sure it will work.

Reply:
Ahh right. You cant do logic before calling the other constructor.

Reply:
I solved it, don't worry